dataservice/extensions/flask_indexd.py

```diff
@@ -224,20 +257,25 @@ def delete(self, record):
             return record
 
         if record.rev is None:
-            r = self.session.get(self.url + record.latest_did)
+            try:
+                r = self.session.get(self.url + record.latest_did)
+            except RequestException as err:
+                abort(500, "Problem getting record in Indexd")
             record.rev = r.json()['rev']
 
         url = '{}{}?rev={}'.format(self.url, record.latest_did, record.rev)
-        resp = self.session.delete(url)
-        self.check_response(resp)
         try:
+            resp = self.session.delete(url)
+            self.check_response(resp)
             resp.raise_for_status()
         except HTTPError:
             message = 'could not get file record'
             # Attach indexd error message, if there is one
             if 'error' in resp.json():
```

Comment:
Hmm why do we only attach the indexd error on delete errors? Seems like a good thing to do for all requests. 

I think its worth considering a common `send_request` method that can be called everywhere you send a request and do exception/error handling. It could take in the requests HTTP method, some request kwargs, and also an optional error message string to return if aborting.